allennlp/commands/predict.py

```diff
@@ -196,7 +223,15 @@ def _get_instance_data(self) -> Iterator[Instance]:
         elif self._dataset_reader is None:
             raise ConfigurationError("To generate instances directly, pass a DatasetReader.")
         else:
-            yield from self._dataset_reader.read(self._input_file)
+            if isinstance(self._dataset_reader, MultiTaskDatasetReader):
+                assert (
```

Comment:
Why do we check this again?

Reply:
I see it more as documentation than as code. It's executable documentation. The next line will fail if that `assert` isn't met. If the `assert` fails, you are already one step closer to figuring out what's wrong than if your error message is something like "Unknown parameter: `force_task`".